src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                return new Square(color, random.nextDouble() * 10);
+            case 1:
+                return new Rectangle(color, random.nextDouble() * 10, random.nextDouble() * 10);
+            case 2:
+                return new RightTriangle(color, random.nextDouble() * 10, random.nextDouble() * 10);
+            case 3:
+                return new Circle(color, random.nextDouble() * 10);
+            case 4:
+                return new IsoscelesTrapezoid(color, random.nextDouble()
+                        * 10, random.nextDouble()
+                        * 10, random.nextDouble()
+                        * 10);
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
```

Comment:
Don't return `null` from a method. It's a good practice to handle all cases within the method itself. In this case, you can put the `getDefaultFigure()` call in the `default` case of your `switch` statement.